packages/web-components/fast-components/src/radio/radio.ts

```diff
@@ -0,0 +1,212 @@
+    keyCodeSpace,
+} from "@microsoft/fast-web-utilities";
+import { FormAssociated } from "../form-associated";
+import { FASTRadio } from ".";
+
+const radioGroups = new Map<string, FASTRadio[]>();
+
+export class Radio extends FormAssociated<HTMLInputElement> {
+    @attr({ attribute: "readonly", mode: "boolean" })
+    public readOnly: boolean; // Map to proxy element
+    private readOnlyChanged(): void {
+        if (this.proxy instanceof HTMLElement) {
+            this.proxy.readOnly = this.readOnly;
+        }
+
+        this.readOnly
+            ? this.classList.add("readonly")
+            : this.classList.remove("readonly");
+    }
+
+    @attr
+    public name: string; // Map to proxy element
+    protected nameChanged(): void {
+        if (this.proxy instanceof HTMLElement) {
+            this.proxy.name = this.name;
+        }
+    }
+
+    /**
+     * The element's value to be included in form submission when checked.
+     * Default to "on" to reach parity with input[type="radio"]
+     */
+    public value: string = "on"; // Map to proxy element.
+    private valueChanged(): void {
+        if (this.proxy instanceof HTMLElement) {
+            this.proxy.value = this.value;
+        }
+    }
+
+    /**
+     * Provides the default checkedness of the input element
+     * Passed down to proxy
+     */
+    @attr({ attribute: "checked", mode: "boolean" })
+    public checkedAttribute: boolean;
+    private checkedAttributeChanged(): void {
+        this.defaultChecked = this.checkedAttribute;
+    }
+
+    /**
+     * Initialized to the value of the checked attribute. Can be changed independently of the "checked" attribute,
+     * but changing the "checked" attribute always additionally sets this value.
+     */
+    @observable
+    public defaultChecked: boolean = !!this.checkedAttribute;
+    private defaultCheckedChanged(): void {
+        if (!this.dirtyChecked) {
+            // Setting this.checked will cause us to enter a dirty state,
+            // but if we are clean when defaultChecked is changed, we want to stay
+            // in a clean state, so reset this.dirtyChecked
+            this.checked = this.defaultChecked;
+            this.dirtyChecked = false;
+        }
+    }
+
+    /**
+     * The checked state of the control
+     */
+    @observable
+    public checked: boolean = this.defaultChecked;
+    private checkedChanged(): void {
+        if (!this.dirtyChecked) {
+            this.dirtyChecked = true;
+        }
+
+        if (this.proxy instanceof HTMLElement) {
+            this.proxy.checked = this.checked;
+        }
+
+        this.dispatchEvent(new CustomEvent("change", { bubbles: true, composed: true }));
+        this.checked ? this.classList.add("checked") : this.classList.remove("checked");
+        this.checkedAttribute = this.checked;
+
+        if (this.checked) {
+            this.updateOtherGroupRadios();
+        }
+    }
+
+    private updateOtherGroupRadios(): void {
```

Comment:
I'm a bit torn here - should a radio be responsible for changing the attributes of other radios or should a radiogroup handle this? I think it's clever, but would a tab manage other tabs, or would the tabs component do that? Is this a principle we'd align to elsewhere?

Reply:
tabs don't have this standard that radio has without our components, people can use the name attribute to group radio buttons and get this behavior by default. Even though I have a name attribute that is something that is normally built in.

Reply:
I think the primary concern here is ensuring that we have a principle that we're applying. Menu items can support several internal roles such as "menuitem", "menuitemradio", "menuitemcheckbox". I think it falls down in that example as the menu items need to be a child of a menu. 

As I said, I'm a bit torn but I understand the intent. All that said, it would've been nice to have this detailed out in the spec (or added after the fact once it was decided). We should document this somewhere - for now, I'd likely recommend adding the details to the spec as we don't have a full story for the Readme files yet.

Reply:
It was stated in the spec that the name attribute was used to group controls. I was under the impression we also wanted to create components that would adhere to html standards as they were, in this case if you required the use of a radiogroup component to get that behavior it wouldn't be the standard way radios are grouped in html. I understand wanting to have consistency, I guess there's a trade off decision to be made there.

Reply:
I think the issue is more ensuring that we cover all bases when we add something like this which seems to be a valuable convenience. The radio behavior is slightly different when inside a toolbar: https://github.com/microsoft/fast-dna/pull/2985/files#r413181301. As the W3C is a standard, we need to consider if we're going to bake the convenience in for that as well or if we expect authors to do something different.

Reply:
I'm going to resolve this conversation as it's similar to the other discussion which I'll leave open until properly addressed.